### DISCUSSION TITLE
Replace doT templates with tempura or another templating engine

### DISCUSSION THREAD

**Discussion by MikeRalphson** [Mermade/widdershins#461] [Ideas]
[doT](https://github.com/olado/doT) is/was maintained by the author of `ajv`, but `ajv` no longer uses `doT` and I suspect it will not get much maintenance going forward.

I have looked at [tempura](https://github.com/lukeed/tempura) from @lukeed which is actively maintained (though new) and appears to be faster than `doT` (though that may not be a major consideration for most `widdershins` users).

The syntax is very similar to handlebars or mustache, but with simple logic flow helpers.

My initial thought is to support but deprecate `doT` templates in the next major version of `widdershins`, and either try to parse templates as `tempura` but fallback to `doT` if compilation fails, or have an option to control the template engine. In the next major version after that, support for `doT` templates would be removed.

Any and all thoughts gratefully received. Thank you.

**Comment by lukeed** [comment]
Oh hey, thanks! Lemme know if you need anything. 

A few people have found the strictness of tempura templates to be an obstacle. I definitely prefer the strictness (aka, need to `#expect` the var names the template will be using), but it does make the transition from handlebars/wherever slightly more difficult. So I'm releasing a new version that now supports [`options.loose`](https://github.com/lukeed/tempura/blob/master/docs/api.md#optionsloose).
